internal/slicer/report_test.go

```diff
@@ -0,0 +1,214 @@
+package slicer_test
+
+import (
+	"io/fs"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/setup"
+	"github.com/canonical/chisel/internal/slicer"
+)
+
+var oneSlice = &setup.Slice{
+	Package:   "base-files",
+	Name:      "my-slice",
+	Essential: nil,
+	Contents:  nil,
+	Scripts:   setup.SliceScripts{},
+}
+
+var otherSlice = &setup.Slice{
+	Package:   "base-files",
+	Name:      "other-slice",
+	Essential: nil,
+	Contents:  nil,
+	Scripts:   setup.SliceScripts{},
+}
+
+var sampleDir = fsutil.Info{
+	Path: "/root/exampleDir",
+	Mode: fs.ModeDir | 0654,
+	Link: "",
+}
+
+var sampleFile = fsutil.Info{
+	Path: "/root/exampleFile",
+	Mode: 0777,
+	Hash: "exampleFile_hash",
+	Size: 5678,
+	Link: "",
+}
+
+var sampleLink = fsutil.Info{
+	Path: "/root/exampleLink",
+	Mode: 0777,
+	Hash: "exampleFile_hash",
+	Size: 5678,
+	Link: "/root/exampleFile",
+}
+
+type sliceAndInfo struct {
+	info  fsutil.Info
+	slice *setup.Slice
+}
+
+var reportTests = []struct {
+	summary      string
+	sliceAndInfo []sliceAndInfo
```

Comment:
This is saying what it is (a slice and an info) but not what the point of it is. Naming the field as `add` would make it more clear, and maybe `entries` instead of `expected`, but that latter one is more reasonable, at least until we have other expectations that we want to check.